tests/dualtor/test_orchagent_slb.py

```diff
@@ -0,0 +1,305 @@
+    vlan_intf_addr = upper_tor_vlan["addr"]
+    vlan_intf_prefixlen = upper_tor_vlan["prefixlen"]
+
+    # construct the server ip with the vlan prefix length
+    upper_tor_server_ip = "%s/%s" % (test_server_ip.split("/")[0], vlan_intf_prefixlen)
+    lower_tor_server_ip = "%s/%s" % (test_server_ip.split("/")[0], vlan_intf_prefixlen)
+
+    # find ToRs' ASNs
+    upper_tor_asn = upper_tor_mg_facts["minigraph_bgp_asn"]
+    lower_tor_asn = lower_tor_mg_facts["minigraph_bgp_asn"]
+    assert upper_tor_asn == lower_tor_asn
+
+    connections = {
+        "upper_tor": {
+            "localhost": upper_tor_host,
+            "local_intf": TEST_DEVICE_INTERFACE,
+            "local_addr": upper_tor_intf_addr,
+            "local_asn": upper_tor_asn,
+            "test_intf": test_iface,
+            "neighbor_intf": upper_tor_server_ptf_intf,
+            "neighbor_addr": upper_tor_server_ip,
+            "neighbor_asn": NEIGHBOR_ASN_UPPER_TOR,
+            "neighbor_type": NEIGHBOR_TYPE,
+            "exabgp_port": EXABGP_PORT_UPPER_TOR,
+        },
+        "lower_tor": {
+            "localhost": lower_tor_host,
+            "local_intf": TEST_DEVICE_INTERFACE,
+            "local_addr": lower_tor_intf_addr,
+            "local_asn": lower_tor_asn,
+            "test_intf": test_iface,
+            "neighbor_intf": lower_tor_server_ptf_intf,
+            "neighbor_addr": lower_tor_server_ip,
+            "neighbor_asn": NEIGHBOR_ASN_LOWER_TOR,
+            "neighbor_type": NEIGHBOR_TYPE,
+            "exabgp_port": EXABGP_PORT_LOWER_TOR,
+        }
+    }
+
+    try:
+        ptfhost.shell("ifconfig %s %s" % (upper_tor_server_ptf_intf, upper_tor_server_ip))
+        for conn in connections.values():
+            ptfhost.shell("ip route add %s via %s" % (conn["local_addr"], vlan_intf_addr))
+        yield connections
+    finally:
+        for conn in connections.values():
+            ptfhost.shell("ifconfig %s 0.0.0.0" % conn["neighbor_intf"], module_ignore_errors=True)
+            ptfhost.shell("ip route del %s" % conn["local_addr"], module_ignore_errors=True)
+
+
+@pytest.fixture
+def bgp_neighbors(upper_tor_host, lower_tor_host, ptfhost, setup_interfaces):
+    """Build the bgp neighbor objects used to start new bgp sessions."""
+    # allow ebgp neighbors that are multiple hops away
+    is_quagga = True
+    connections = setup_interfaces
+    neighbors = {}
+    for dut, conn in connections.items():
+        neighbors[dut] = bgp.BGPNeighbor(
+            conn["localhost"],
+            ptfhost,
+            "slb_%s" % dut,
+            conn["neighbor_addr"].split("/")[0],
+            conn["neighbor_asn"],
+            conn["local_addr"].split("/")[0],
+            conn["local_asn"],
+            conn["exabgp_port"],
+            conn["neighbor_type"],
+            is_quagga=is_quagga
+        )
+    return neighbors
+
+
+@pytest.fixture
+def constants(setup_interfaces):
+    class _C(object):
```

Comment:
Why do we need to create a dummy class as opposed to using a dictionary to store this info?

Reply:
Just prefer using '.' usage.